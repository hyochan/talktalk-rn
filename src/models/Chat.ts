import { User } from './User';

export class Chat {
  public constructor(
    public id: string = '', 
    public sender: User = undefined, 
    public message: string = '', 
    public created: Date = undefined, 
    public updated: Date = undefined
  ) {}
}
