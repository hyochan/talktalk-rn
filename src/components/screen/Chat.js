// @flow
import React, {
  Component,
  useState,
  useEffect,
  useRef,
} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  Keyboard,
  Platform,
  TextInput,
  KeyboardAvoidingView,
  FlatList,
} from 'react-native';
import styled from 'styled-components/native';

import { Header } from 'react-navigation';

import type {
  ____ViewStyleProp_Internal as ViewStyle,
  ____TextStyleProp_Internal as TextStyle,
  ____ImageStyleProp_Internal as ImageStyle,
} from 'react-native/Libraries/StyleSheet/StyleSheetTypes';

import ChatListItem from '../shared/ChatListItem';
import EmptyListItem from '../shared/EmptyListItem';
import Button from '../shared/Button';

import { Chatroom } from '../../models/Chatroom';
import { IC_SMILE } from '../../utils/Icons';
import { ratio, colors } from '../../utils/Styles';
import { getString } from '../../../STRINGS';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;

  flex-direction: column;
  align-items: center;
`;

const StyledViewChat = styled.View`
  width: 100%;
  border-top-width: 0.5;
  border-color: rgb(225,225,225);
  background-color: white;
  min-height: 52;
  max-height: 100;

  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
`;

const StyledInputChat = styled.TextInput`
  width: 80%;
  font-size: 14;
  margin-right: 20;
  padding-left: 48;
  color: black;
`;

const StyledTouchMenu = styled.TouchableOpacity`
  position: absolute;
  left: 10;
  height: 100%;
  minWidth: 20;
  justifyContent: center;
`;

const StyledViewBottom = styled.View`
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const StyledViewMenu = styled.View`
  height: 258;
  background-color: green;
`;

type Styles = {
  btnSend: ViewStyle,
  txtSend: TextStyle,
};

const styles: Styles = StyleSheet.create({
  btnSend: {
    right: 8,
    backgroundColor: colors.dodgerBlue,
    borderRadius: 4,
    width: 60,
    height: 36,

    alignItems: 'center',
    justifyContent: 'center',
  },
  txtSend: {
    fontSize: 14,
    fontWeight: 'bold',
    color: 'white',
    paddingHorizontal: 5,
    paddingVertical: 10,
  },
});

type Props = {
  navigation: any;
};
type State = {
  isLoading: boolean,
  showMenu: boolean,
  message: string,
  chats: Chat[],
};

function Screen(props: Props, state: State) {
  let keyboardDidShowListener: any;
  let keyboardDidHideListener: any;

  const input1 = useRef(null);
  const input2 = useRef(null);
  const [isLoading, setIsLoading] = useState(false);
  const [showMenu, setShowMenu] = useState(false);
  const [message, setMessage] = useState('');
  const [chats, setChats] = useState([
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'hello1',
    ),
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'Hello2. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.',
    ),
    new Chat(
      '',
      new User(
        '1',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.',
    ),
  ]);

  useEffect(() => {
    if (showMenu) {
      Keyboard.dismiss();
    } else {
      if (input1 && input1.current) {
        input1.current.focus();
      }
    }
  }, [showMenu]);

  useEffect(() => {
    Keyboard.addListener('keyboardDidShow', (e) => {
      console.log('keyboardHeight', e.endCoordinates.height);
    });
    return () => {
      if (keyboardDidShowListener) {
        keyboardDidShowListener.remove();
      }
    };
  });

  const renderItem = ({ item, index } : any) => {
    return (
      <ChatListItem
        prevItem={index > 0 ? chats[ index - 1 ] : null}
        item={item}
      />
    );
  };

  const sendChat = () => {
    console.log('sendChat');
  };

  const goBack = () => {
    props.navigation.goBack();
  };

  return (
    <StyledContainer>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset = {Header.HEIGHT + 24}
        behavior={Platform.select({
          'android': null,
          'ios': 'padding',
        })}
      >
        <FlatList
          style={{ alignSelf: 'stretch' }}
          contentContainerStyle={
            chats.length === 0
              ? {
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={chats}
          renderItem={renderItem}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
        {
          !showMenu
            ? <StyledViewChat>
              <StyledInputChat
                ref={input1}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={ colors.cloudyBlue }
                value={message}
                defaultValue={message}
                onChangeText={(text) => setMessage(text)}
              />
              <StyledTouchMenu
                onPress={() => setShowMenu(true)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                isLoading={isLoading}
                onPress={sendChat}
                style={styles.btnSend}
                textStyle={styles.txtSend}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            : null
        }
      </StyledKeyboardAvoidingView>
      {
        showMenu
          ? <StyledViewBottom>
            <StyledViewChat>
              <StyledInputChat
                ref={input2}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={ colors.cloudyBlue }
                value={message}
                defaultValue={message}
              />
              <StyledTouchMenu
                onPress={() => setShowMenu(false)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                isLoading={isLoading}
                onPress={sendChat}
                style={styles.btnSend}
                textStyle={styles.txtSend}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            <StyledViewMenu />
          </StyledViewBottom>
          : null
      }
    </StyledContainer>
  );
}

export default Screen;
