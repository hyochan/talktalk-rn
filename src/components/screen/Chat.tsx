import React, {
  Component,
  useState,
  useEffect,
  useRef,
} from 'react';
import {
  SafeAreaView,
  StyleSheet,
  TouchableOpacity,
  Image,
  Text,
  View,
  Keyboard,
  Platform,
  TextInput,
  KeyboardAvoidingView,
  FlatList,
} from 'react-native';
import { NavigationScreenProp, NavigationStateRoute } from 'react-navigation';
import styled, { ThemeProps, withTheme, DefaultTheme } from 'styled-components/native';

import { Header } from 'react-navigation';
import Icon5, { FA5Style } from 'react-native-vector-icons/FontAwesome5';
import ChatListItem from '../shared/ChatListItem';
import EmptyListItem from '../shared/EmptyListItem';
import Button from '../shared/Button';

import { Chatroom } from '../../models/Chatroom';
import { IC_SMILE } from '../../utils/Icons';
import { getString } from '../../../STRINGS';
import { Chat } from '../../models/Chat';
import { User } from '../../models/User';

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background-color: white;
  flex-direction: column;
  align-items: center;
`;

const StyledKeyboardAvoidingView = styled.KeyboardAvoidingView`
  flex: 1;
  justify-content: center;
  align-self: stretch;
  flex-direction: column;
  align-items: center;
`;

const StyledViewChat = styled.View`
  width: 100%;
  border-top-width: 0.5;
  border-color: rgb(225,225,225);
  background-color: white;
  min-height: 52;
  max-height: 100;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
`;

const StyledInputChat = styled.TextInput`
  width: 80%;
  font-size: 14;
  margin-right: 20;
  padding-left: 48;
  color: black;
`;

const StyledTouchMenu = styled.TouchableOpacity`
  position: absolute;
  left: 10;
  height: 100%;
  min-width: 20px;
  justify-content: center;
`;

const StyledViewBottom = styled.View`
  position: absolute;
  bottom: 0;
  width: 100%;
`;

const StyledViewMenu = styled.View<{ height: number }>`
  background-color: ${({ theme: { colors: { paleGray }}}) => paleGray};
  flex-direction: row;
  flex-wrap: wrap;
  height: ${({ height }) => height};
`;

interface IProps extends ThemeProps<DefaultTheme> {
  navigation: any;
}

interface IState {
  isLoading: boolean;
  showMenu: boolean;
  message: string;
  chats: Chat[];
}

function Screen(props: IProps, state: IState) {
  let keyboardShowListener: any;

  const input1 = useRef(null);
  const input2 = useRef(null);
  const [keyboardHeight, setKeyboardHeight] = useState(258);
  const [isLoading, setIsLoading] = useState(false);
  const [showMenu, setShowMenu] = useState(false);
  const [message, setMessage] = useState('');
  const [chats, setChats] = useState([
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'hello1',
    ),
    new Chat(
      '',
      new User(
        '0',
        'sender111',
        '',
      ),
      'Hello2. This is long message. This is long message. This is long message.This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.',
    ),
    new Chat(
      '',
      new User(
        '1',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello',
    ),
    new Chat(
      '',
      new User(
        '2',
        'sender2222',
        '',
      ),
      'hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.',
    ),
  ]);

  useEffect(() => {
    if (showMenu) {
      Keyboard.dismiss();
    } else {
      if (input1 && input1.current) {
        input1.current.focus();
      }
    }
  }, [showMenu]);

  useEffect(() => {
    keyboardShowListener = Keyboard.addListener('keyboardDidShow', (e) => {
      setKeyboardHeight(e.endCoordinates.height);
    });
    return () => {
      if (keyboardShowListener) {
        keyboardShowListener.remove();
      }
    };
  });

  const renderItem = ({ item, index }: any) => {
    return (
      <ChatListItem
        prevItem={index > 0 ? chats[ index - 1 ] : null}
        item={item}
      />
    );
  };

  const sendChat = () => {
  };

  const {
    theme: {
      colors: {
        dusk,
        cloudyBlue
      }
    }    
  } = props;

  return (
    <StyledContainer>
      <StyledKeyboardAvoidingView
        keyboardVerticalOffset={Header.HEIGHT}
        behavior={Platform.select({
          android: null,
          ios: 'padding',
        })}
      >
        <FlatList
          style={{ alignSelf: 'stretch' }}
          contentContainerStyle={
            chats.length === 0
              ? {
                flex: 1,
                alignItems: 'center',
                justifyContent: 'center',
              }
              : null
          }
          keyExtractor={(item, index) => index.toString()}
          data={chats}
          renderItem={renderItem}
          ListEmptyComponent={<EmptyListItem>{getString('NO_CONTENT')}</EmptyListItem>}
        />
        {
          !showMenu
            ? <StyledViewChat>
              <StyledInputChat
                testID='input_chat'
                ref={input1}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={cloudyBlue}
                value={message}
                defaultValue={message}
                onChangeText={(text) => setMessage(text)}
              />
              <StyledTouchMenu
                testID='touch_menu'
                onPress={() => setShowMenu(true)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                testID='btn_chat'
                isLoading={isLoading}
                onPress={sendChat}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            : null
        }
      </StyledKeyboardAvoidingView>
      {
        showMenu
          ? <StyledViewBottom>
            <StyledViewChat>
              <StyledInputChat
                ref={input2}
                onFocus={() => setShowMenu(false)}
                multiline={true}
                placeholder={ getString('WRITE_MESSAGE') }
                placeholderTextColor={cloudyBlue}
                value={message}
                defaultValue={message}
              />
              <StyledTouchMenu
                testID='touch_menu'
                onPress={() => setShowMenu(false)}
              >
                <Image style={{
                  width: 20,
                  height: 20,
                }} source={IC_SMILE}/>
              </StyledTouchMenu>
              <Button
                testID='btn_chat'
                isLoading={isLoading}
                onPress={sendChat}
              >{getString('SEND')}</Button>
            </StyledViewChat>
            <StyledViewMenu testID="viewMenu" height={keyboardHeight}>
              <View style={{
                flexDirection: 'row',
                marginTop: 10,
              }}>
                <TouchableOpacity
                  style={{
                    marginLeft: 16,
                    marginTop: 2,
                    width: 60,
                    height: 60,
                    justifyContent: 'center',
                    alignItems: 'center',
                  }}
                >
                  <Icon5 name='camera' size={36} color={dusk} light />
                </TouchableOpacity>
                <TouchableOpacity
                  style={{
                    marginLeft: 16,
                    marginTop: 4,
                    width: 60,
                    height: 60,
                    justifyContent: 'center',
                    alignItems: 'center',
                  }}
                >
                  <Icon5 name='image' size={40} color={dusk} light/>
                </TouchableOpacity>
              </View>
            </StyledViewMenu>
          </StyledViewBottom>
          : null
      }
    </StyledContainer>
  );
}

export default withTheme(Screen);
