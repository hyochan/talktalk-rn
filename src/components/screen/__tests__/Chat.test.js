import 'react-native';
import * as React from 'react';
import Chat, { styles } from '../Chat';
import  { getString } from '../../../../STRINGS';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

const props = {
  navigation: {
    navigate: jest.fn(),
  },
};
const component: React.Element<any> = <Chat {...props}/>;

describe('[Chat] rendering test', () => {
  it('renders as expected', () => {
    const json = renderer.create(component).toJSON();
    expect(json).toMatchSnapshot();
  });
});

describe('[Login] interaction', () => {
  let rendered: renderer.ReactTestRenderer;
  let root: renderer.ReactTestInstance;
  let testingLib: any;

  beforeAll(() => {
    rendered = renderer.create(component);
    root = rendered.root;
    testingLib = render(component);
  });

  it('should invoke changeText event handler when message changed', () => {
    const textInput = testingLib.getByTestId('input_chat');
    jest.useFakeTimers();
    jest.runAllTimers();
    fireEvent.changeText(textInput, 'chat test');
    expect(textInput.props.value).toEqual('chat test');
  });

  it('should call [setShowMenu] when focused', () => {
    const textInput = testingLib.getByTestId('input_chat');
    textInput.props.onFocus();
  });

  it('should [showMenu] when touch pressed', () => {
    let touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent(touchMenu, 'press');

    touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent(touchMenu, 'press');
  });

  it('should call [setShowMenu] when focused', () => {
    const touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent(touchMenu, 'press');
  });

  it('should [sendChat] when pressing button', () => {
    let chatBtn = testingLib.getByTestId('btn_chat');
    fireEvent(chatBtn, 'press');

    const touchMenu = testingLib.getByTestId('touch_menu');
    fireEvent(touchMenu, 'press');

    chatBtn = testingLib.getByTestId('btn_chat');
    fireEvent(chatBtn, 'press');
  });
});