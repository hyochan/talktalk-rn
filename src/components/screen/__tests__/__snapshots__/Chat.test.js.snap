// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Chat] rendering test renders as expected 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexBasis": 0,
        "flexDirection": "column",
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexBasis": 0,
            "flexDirection": "column",
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
        ],
        Object {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <RCTScrollView
      ListEmptyComponent={
        <Shared
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flex": 1,
              "justifyContent": "center",
              "width": "100%",
            }
          }
        >
          No content
        </Shared>
      }
      contentContainerStyle={null}
      data={
        Array [
          Chat {
            "created": undefined,
            "id": "",
            "message": "hello1",
            "sender": User {
              "Chatrooms": Array [],
              "chatrooms": undefined,
              "created": undefined,
              "displayName": "sender111",
              "friends": undefined,
              "isOnline": undefined,
              "photoURL": "",
              "statusMsg": undefined,
              "uid": "0",
              "updated": undefined,
            },
            "updated": undefined,
          },
          Chat {
            "created": undefined,
            "id": "",
            "message": "Hello2. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.",
            "sender": User {
              "Chatrooms": Array [],
              "chatrooms": undefined,
              "created": undefined,
              "displayName": "sender111",
              "friends": undefined,
              "isOnline": undefined,
              "photoURL": "",
              "statusMsg": undefined,
              "uid": "0",
              "updated": undefined,
            },
            "updated": undefined,
          },
          Chat {
            "created": undefined,
            "id": "",
            "message": "hello",
            "sender": User {
              "Chatrooms": Array [],
              "chatrooms": undefined,
              "created": undefined,
              "displayName": "sender2222",
              "friends": undefined,
              "isOnline": undefined,
              "photoURL": "",
              "statusMsg": undefined,
              "uid": "1",
              "updated": undefined,
            },
            "updated": undefined,
          },
          Chat {
            "created": undefined,
            "id": "",
            "message": "hello",
            "sender": User {
              "Chatrooms": Array [],
              "chatrooms": undefined,
              "created": undefined,
              "displayName": "sender2222",
              "friends": undefined,
              "isOnline": undefined,
              "photoURL": "",
              "statusMsg": undefined,
              "uid": "2",
              "updated": undefined,
            },
            "updated": undefined,
          },
          Chat {
            "created": undefined,
            "id": "",
            "message": "hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.",
            "sender": User {
              "Chatrooms": Array [],
              "chatrooms": undefined,
              "created": undefined,
              "displayName": "sender2222",
              "friends": undefined,
              "isOnline": undefined,
              "photoURL": "",
              "statusMsg": undefined,
              "uid": "2",
              "updated": undefined,
            },
            "updated": undefined,
          },
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "alignSelf": "stretch",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginLeft": 20,
                  "marginRight": 8,
                  "minHeight": 48,
                },
                Object {
                  "marginRight": 80,
                  "marginTop": 20,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="peer_image"
              >
                <Unknown
                  color="rgb(65,77,107)"
                  light={true}
                  name="meh"
                  size={40}
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "marginBottom": 2,
                    },
                  ]
                }
              >
                sender111
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "rgb(225,225,225)",
                    "borderRadius": 3,
                    "borderWidth": 1,
                    "marginRight": 8,
                    "marginVertical": 4,
                    "padding": 12,
                    "shadowColor": "rgb(233,237,244)",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgb(65,77,107)",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  hello1
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(175,194,219)",
                      "fontSize": 12,
                      "marginRight": 20,
                    },
                  ]
                }
              >
                0 : 0
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginLeft": 20,
                  "marginRight": 8,
                  "minHeight": 48,
                },
                Object {
                  "marginRight": 80,
                  "marginTop": 0,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="peer_image"
              >
                <View
                  style={
                    Object {
                      "width": 40,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <View />
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "rgb(225,225,225)",
                    "borderRadius": 3,
                    "borderWidth": 1,
                    "marginRight": 8,
                    "marginVertical": 4,
                    "padding": 12,
                    "shadowColor": "rgb(233,237,244)",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgb(65,77,107)",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  Hello2. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message. This is long message.
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(175,194,219)",
                      "fontSize": 12,
                      "marginRight": 20,
                    },
                  ]
                }
              >
                0 : 0
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-start",
                  "flexDirection": "row",
                  "justifyContent": "flex-start",
                  "marginLeft": 20,
                  "marginRight": 8,
                  "minHeight": 48,
                },
                Object {
                  "marginRight": 80,
                  "marginTop": 20,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "marginRight": 8,
                }
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="peer_image"
              >
                <Unknown
                  color="rgb(65,77,107)"
                  light={true}
                  name="meh"
                  size={40}
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "flexDirection": "column",
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(65,77,107)",
                      "fontSize": 12,
                      "marginBottom": 2,
                    },
                  ]
                }
              >
                sender2222
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "white",
                    "borderColor": "rgb(225,225,225)",
                    "borderRadius": 3,
                    "borderWidth": 1,
                    "marginRight": 8,
                    "marginVertical": 4,
                    "padding": 12,
                    "shadowColor": "rgb(233,237,244)",
                    "shadowOffset": Object {
                      "height": 2,
                      "width": 0,
                    },
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "rgb(65,77,107)",
                        "fontSize": 14,
                      },
                    ]
                  }
                >
                  hello
                </Text>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(175,194,219)",
                      "fontSize": 12,
                      "marginRight": 20,
                    },
                  ]
                }
              >
                0 : 0
              </Text>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                  "marginTop": 20,
                  "minHeight": 48,
                  "width": "100%",
                },
              ]
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4284782804,
                  4282027007,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0.8,
                }
              }
              locations={
                Array [
                  0,
                  0.85,
                ]
              }
              startPoint={
                Object {
                  "x": 0.2,
                  "y": 0.4,
                }
              }
              style={
                Object {
                  "borderRadius": 3,
                  "marginHorizontal": 20,
                  "marginLeft": 8,
                  "padding": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                hello
              </Text>
            </BVLinearGradient>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgb(175,194,219),",
                    "fontSize": 12,
                    "marginRight": 20,
                    "marginTop": 4,
                  },
                ]
              }
            >
              0 : 0
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flexDirection": "column",
                  "justifyContent": "flex-end",
                  "marginTop": 20,
                  "minHeight": 48,
                  "width": "100%",
                },
              ]
            }
          >
            <BVLinearGradient
              colors={
                Array [
                  4284782804,
                  4282027007,
                ]
              }
              endPoint={
                Object {
                  "x": 1,
                  "y": 0.8,
                }
              }
              locations={
                Array [
                  0,
                  0.85,
                ]
              }
              startPoint={
                Object {
                  "x": 0.2,
                  "y": 0.4,
                }
              }
              style={
                Object {
                  "borderRadius": 3,
                  "marginHorizontal": 20,
                  "marginLeft": 8,
                  "padding": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 14,
                    },
                  ]
                }
              >
                hello. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message. This is my long message.
              </Text>
            </BVLinearGradient>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgb(175,194,219),",
                    "fontSize": 12,
                    "marginRight": 20,
                    "marginTop": 4,
                  },
                ]
              }
            >
              0 : 0
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderColor": "rgb(225,225,225)",
            "borderTopWidth": 0.5,
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "maxHeight": 100,
            "minHeight": 52,
            "width": "100%",
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        defaultValue=""
        multiline={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Write message..."
        placeholderTextColor="rgb(175,194,219)"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "black",
              "fontSize": 14,
              "marginRight": 20,
              "paddingLeft": 48,
              "width": "80%",
            },
          ]
        }
        testID="input_chat"
        underlineColorAndroid="transparent"
        value=""
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": "100%",
            "justifyContent": "center",
            "left": 10,
            "minWidth": 20,
            "opacity": 1,
            "position": "absolute",
          }
        }
        testID="touch_menu"
      >
        <Image
          source={
            Object {
              "testUri": "../../../assets/icons/icSmile.png",
            }
          }
          style={
            Object {
              "height": 20,
              "width": 20,
            }
          }
        />
      </View>
      <View
        style={
          Array [
            Object {
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="press_id"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(58,139,255)",
                  "borderColor": "rgb(58,139,255)",
                  "borderRadius": 4,
                  "borderWidth": 1,
                  "justifyContent": "center",
                  "shadowColor": "rgb(58,139,255)",
                  "shadowOpacity": 0.3,
                  "shadowRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "height": 60,
                  "justifyContent": "center",
                  "width": 136,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "white",
                      "fontSize": 14,
                      "fontWeight": "bold",
                    },
                  ]
                }
              >
                Send
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
